src/main.rs

```diff
@@ -67,7 +67,7 @@ fn nand(a: bool, b: bool) -> bool {
     !(a && b)
 }
 
-fn get_link_by_id(store: &mut unit::Store<usize, _>, id: usize) -> Result<Link<usize>, Error<usize>> {
+fn get_link_by_id<T>(store: &mut unit::Store<usize, T>, id: usize) -> Result<Link<usize>, Error<usize>> {
```

Comment:
If you add a type parameter also update all usages of this function to be correct. Use `LinkType` instead of `Link` here.